activerecord/lib/active_record/relation/merger.rb

```diff
@@ -187,6 +192,21 @@ def replace_from_clause?
           relation.from_clause.empty? && !other.from_clause.empty? &&
             relation.model.base_class == other.model.base_class
         end
+
+        def raw_sql_involved?(where_clause1, where_clause2)
+          where_clause1.ast.grep(Arel::Nodes::SqlLiteral).any? || where_clause2.ast.grep(Arel::Nodes::SqlLiteral).any?
+        end
+
+        def merge_with_proper_or_handling(where_clause1, where_clause2)
+          combined_conditions = Arel::Nodes::And.new(
```

Comment:
I'm afraid using and instead of merge is not the solution; In rails 7 the behaviour of merge diverged from and, see for instance:

https://www.fastruby.io/blog/rails/upgrades/rails-merge-deprecation.html